test/hotspot/jtreg/compiler/c2/irTests/TestShiftAndMask.java

```diff
@@ -120,7 +120,7 @@ public static void checkShiftNonConstMaskLong(long res) {
     @IR(counts = { IRNode.AND_I, "1" })
     @IR(failOn = { IRNode.ADD_I, IRNode.LSHIFT_I })
     public static int addShiftMaskInt(int i, int j) {
-        return (j + (i << 2)) & 3; // transformed to: return j & 3;
+        return (j + ((i + 1) << 2)) & 3; // transformed to: return j & 3;
```

Comment:
I would prefer you adding other test cases instead of modifying existing ones.

Reply:
Done.